### ISSUE TITLE
do not fire `value-changed` when disconnecting from the DOM

### ISSUE THREAD

**Issue by vlukashov** [vaadin/vaadin-dropdown-menu#142]
Suppose I have a component with a dropdown menu in its shadow DOM. I have added a `value-changed` event listener to the dropdown. I do not expect this listener to get invoked when my component is removed from the page, yet this is what actually happens.

```html
<vaadin-dropdown-menu value="{{language}}" label="[[label]]">
  <template>
    <vaadin-list-box>
      <template is="dom-repeat" items="[[languages]]">
        <vaadin-item value="[[item.name]]">[[item.label]]</vaadin-item>
      </template>
    </vaadin-list-box>
  </template>
</vaadin-dropdown-menu>
```

```javascript
let languageSelector = this.shadowRoot.querySelector('vaadin-dropdown-menu');
languageSelector.addEventListener('value-changed', e => {
  // This listener is called when the component is disconnected from the DOM
  this.dispatch('setLocale', languageSelector.value);
});
```

**Comment by tomivirkki** [comment]
IIRC this is fixed in master (verify). If this concerns Vaadin 10, cherry pick the fix to "vaadin-10" branch and release maintenance version.

**Comment by web-padawan** [comment]
Fixed in #137, will cherry-pick to "vaadin-10"

**Comment by web-padawan** [comment]
Cherry-picked in 21603e7 and needs a patch release (1.0.1)
